File: backend/src/types/orderbook.ts
export interface IndividualEntry{
  type: 'sell' | 'reverted',
  quantity: number
}

export interface OrderEntry {
  total: number;
  orders: Record<string, IndividualEntry>;
}

export interface Orderbook {
  yes: Record<number, OrderEntry>;
  no: Record<number, OrderEntry>;
}

File: go_worker/models/orderbook.go
package models

type Order struct {
	Type     string `json:"type,omitempty"`
	Quantity int    `json:"quantity,omitempty"`
}

type PriceLevel struct {
	Total  int              `json:"total"`
	Orders map[string]Order `json:"orders"`
}

type OrderBookSymbol struct {
	Yes map[float64]PriceLevel `json:"yes"`
	No  map[float64]PriceLevel `json:"no"`
}

var ORDERBOOK = map[string]OrderBookSymbol{}
